access/services/project.go

```diff
@@ -0,0 +1,165 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+)
+
+const projectsApi = "v1/projects"
+
+type ProjectParams struct {
+	ProjectDetails Project
+}
+
+func NewProjectParams() ProjectParams {
+	return ProjectParams{}
+}
+
+type Project struct {
+	DisplayName       string           `json:"display_name,omitempty"`
+	Description       string           `json:"description,omitempty"`
+	AdminPrivileges   *AdminPrivileges `json:"admin_privileges,omitempty"`
+	SoftLimit         bool             `json:"soft_limit,omitempty"`
+	StorageQuotaBytes float64          `json:"storage_quota_bytes,omitempty"`
+	ProjectKey        string           `json:"project_key,omitempty"`
+}
+
+type AdminPrivileges struct {
+	ManageMembers   bool `json:"manage_members,omitempty"`
+	ManageResources bool `json:"manage_resources,omitempty"`
+	IndexResources  bool `json:"index_resources,omitempty"`
+}
+
+type ProjectService struct {
+	client         *jfroghttpclient.JfrogHttpClient
+	ServiceDetails auth.ServiceDetails
+}
+
+func NewProjectService(client *jfroghttpclient.JfrogHttpClient) *ProjectService {
+	return &ProjectService{client: client}
+}
+
+func (us *ProjectService) SetArtifactoryDetails(rt auth.ServiceDetails) {
+	us.ServiceDetails = rt
+}
+
+func (us *ProjectService) getProjectsBaseUrl() string {
+	return fmt.Sprintf("%s%s", us.ServiceDetails.GetUrl(), projectsApi)
+}
+
+func (us *ProjectService) GetProject(projectKey string) (u *Project, err error) {
+	httpDetails := us.ServiceDetails.CreateHttpClientDetails()
+	url := fmt.Sprintf("%s/%s", us.getProjectsBaseUrl(), projectKey)
+	resp, body, _, err := us.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	// The case the requested user is not found
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+	if err = errorutils.CheckResponseStatus(resp, http.StatusOK); err != nil {
+		return nil, errorutils.CheckError(errorutils.GenerateResponseError(resp.Status, clientutils.IndentJson(body)))
+	}
+	var project Project
+	if err := json.Unmarshal(body, &project); err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	return &project, nil
+}
+
+func (us *ProjectService) CreateProject(params ProjectParams) error {
```

Comment:
This method and the UpdateProject method share most of their code. Let's reuse the code, and have this method call the UpdateProject method after the "project not exists" validation part.